### PR TITLE
coreboot test intergration

### PR DESCRIPTION
**Pull request by sylv-io**
### Add coreboot test integration

__overview:__
 * 1-11: move  necessary functions from `fwts_klog.c` to `fwts_log_scan.c`
 * 12-16: add functions to `fwts_clog.c`
 * 17: add clog test
 * 18: add `--coreboot-log` as argument to parse a coreboot log file
 * 19: skip coreboot test if not available
 * 20: add some coreboot messages to `data/clog.json`

### CODE REVIEW

File: src/lib/src/fwts_clog.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+void fwts_clog_free(fwts_list *clog)
+{
+        fwts_log_free(clog);
+}
+
+bool fwts_clog_available(fwts_framework *fw)
+{
+    char *vendor;
+
+    if (fw->clog || ((vendor = fwts_get("/sys/class/dmi/id/bios_vendor")) && strstr(vendor, COREBOOT_BIOS_VENDOR)))
```
Comment (zaolin) [inline src/lib/src/fwts_clog.c]: standalone vars, do not return in if clauses

File: src/lib/src/fwts_framework.c [hunk 2]
```diff
@@ -135,6 +135,7 @@ static fwts_option fwts_framework_options[] = {
 	{ "arch",		"",   1, "Specify arch of the tables being tested (defaults to current host)." },
 	{ "sbbr",		"",   0, "Run ARM SBBR tests." },
 	{ "ifv",		"",   0, "Run tests in firmware-vendor modes." },
+	{ "coreboot-log",	"",   1, "Specify a coreboot logfile dump" },
```
Comment (zaolin) [inline src/lib/src/fwts_framework.c]: cb-log is shorter

Comment (zaolin) [reply src/lib/src/fwts_framework.c]: oder clog

Comment (sylv-io) [reply src/lib/src/fwts_framework.c]: done
changed to --clog

File: src/lib/src/fwts_clog.c (hunk 1 above)
Comment (sylv-io) [reply src/lib/src/fwts_clog.c]: done

**Pull request merged**